src/protocolgame.cpp

```diff
@@ -403,13 +403,31 @@ void ProtocolGame::parsePacket(NetworkMessage& msg)
 	}
 
 	//a dead player can not performs actions
-	if (player->isRemoved() || player->getHealth() <= 0) {
+	if (player->isDead() || player->getHealth() <= 0) {
+		if (recvbyte == 0x14) {
+			disconnect();
+			addGameTask(&Game::removeCreature, player, false, true);
+			return;
+		}
+
 		if (recvbyte == 0x0F) {
+			if (!player->spawn()) {
```

Comment:
Should not this method be executed on Dispatcher thread? like all other methods which interact with game?
I would move whole logic to Game and then execute with dispatcher maybe.

Reply:
yeah, it should probably, hence the problems with lua stack from @MillhioreBT.
I will make new method for this in game class.